File: APPTOPI/app/src/main/java/io/topi/apptopi/service/ServiceConfig.java
package io.topi.apptopi.service;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ServiceConfig {

    public static Retrofit getRetrofit(){
        String url = "https://api.github.com/search/";

        OkHttpClient client = new OkHttpClient.Builder().connectTimeout(25, TimeUnit.SECONDS)
                .writeTimeout(25, TimeUnit.SECONDS).connectTimeout(25, TimeUnit.SECONDS)
                .retryOnConnectionFailure(true).addInterceptor(new Interceptor() {
                    @Override
                    public Response intercept(Chain chain) throws IOException {
                        Request original = chain.request();
                        HttpUrl originalHttpUrl = original.url();
                        HttpUrl url = originalHttpUrl.newBuilder().build();
                        Request.Builder requestBuilder = original.newBuilder().url(url);
                        Request request = requestBuilder.build();
                        return chain.proceed(request);
                    }
                }).build();

        Retrofit retrofitClient = new Retrofit.Builder()
                .baseUrl(url)
                .addConverterFactory(GsonConverterFactory.create())
                .client(client)
                .build();

        return retrofitClient;
    }
}

File: APPTOPI/app/src/main/java/io/topi/apptopi/main/repositorio/ServiceConfig.kt
package io.topi.apptopi.main.repositorio

import okhttp3.HttpUrl
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.Request
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

class ServiceConfig {

    companion object Factory {
        fun getRetrofit(): Retrofit {
            val url: String = "https://api.github.com/search/";
            var client: OkHttpClient = OkHttpClient.Builder().connectTimeout(25, TimeUnit.SECONDS)
                    .writeTimeout(25, TimeUnit.SECONDS).connectTimeout(25, TimeUnit.SECONDS)
                    .retryOnConnectionFailure(true).addInterceptor { chain ->
                        var original: Request = chain.request()
                        var originalHttpUrl: HttpUrl = original.url()
                        var url: HttpUrl = originalHttpUrl.newBuilder().build()
                        var requestBuilder: Request.Builder = original.newBuilder().url(url)
                        var request: Request = requestBuilder.build()
                        chain.proceed(request)
                    }.build()

            var retrofitClient: Retrofit = Retrofit.Builder()
                    .baseUrl(url)
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(client)
                    .build()

            return retrofitClient
        }
    }
}